test/ResourceManager.test.cpp

```diff
@@ -17,3 +18,7 @@ TEST(ResourceManager, IgnoreNonFilenamesOnConstruction) {
 	// Attempt to load Resource Manager where directories exist with .vol and .clm 'extensions'
 	EXPECT_NO_THROW(ResourceManager resourceManager("./data"));
 }
+
+TEST(ResourceManager, AbortsWithCreationOnFilename) {
+	EXPECT_THROW(ResourceManager("./data/Directory.clm/.keep"), std::runtime_error);
```

Comment:
I was going to say this test name is a little confusing, though I see it was changed in a later commit. I really had to think about this one. I assume it's throwing during construction because you have to pass a folder name, but `.keep` is not a folder name. This is perhaps made more confusing because of the special semantics around the name `.keep`, which is being conveniently re-used for another purpose here.